examples/grpc-service/src/main/java/example/armeria/grpc/HelloServiceImpl.java

```diff
@@ -0,0 +1,158 @@
+     */
+    @Override
+    public void lazyHello(HelloRequest request, StreamObserver<HelloReply> responseObserver) {
+        // You can use the event loop for scheduling a task.
+        RequestContext.current().contextAwareEventLoop().schedule(() -> {
+            responseObserver.onNext(buildReply(toMessage(request.getName())));
+            responseObserver.onCompleted();
+        }, 3, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Sends a {@link HelloReply} using {@code blockingTaskExecutor}.
+     *
+     * @see <a href="https://line.github.io/armeria/server-grpc.html#blocking-service-implementation">Blocking
+     *      service implementation</a>
+     */
+    @Override
+    public void blockingHello(HelloRequest request, StreamObserver<HelloReply> responseObserver) {
+        // Unlike upstream gRPC-Java, Armeria does not run service logic in a separate thread pool by default.
+        // Therefore, this method will run in the event loop, which means that you can suffer the performance
+        // degradation if you call a blocking API in this method. In this case, you have the following options:
+        //
+        // 1. Call a blocking API in the blockingTaskExecutor provided by Armeria.
+        // 2. Set GrpcServiceBuilder.useBlockingTaskExecutor(true) when building your GrpcService.
+        // 3. Call a blocking API in the separate thread pool you manage.
+        //
+        // In this example, we chose the option 1:
+        final ServiceRequestContext ctx = RequestContext.current();
+        ctx.blockingTaskExecutor().submit(() -> {
+            try {
+                // Simulate a blocking API call.
+                Thread.sleep(3000);
+            } catch (Exception ignored) {
+                // Do nothing.
+            }
+            responseObserver.onNext(buildReply(toMessage(request.getName())));
+            responseObserver.onCompleted();
+        });
+    }
+
+    /**
+     * Sends 5 {@link HelloReply} responses when receiving a request.
+     *
+     * @see #lazyHello(HelloRequest, StreamObserver)
+     */
+    @Override
+    public void lotsOfReplies(HelloRequest request, StreamObserver<HelloReply> responseObserver) {
+        // You can also write this code without Reactor like 'lazyHello' example.
+        Flux.interval(Duration.ofSeconds(1))
+            .take(5)
+            .map(index -> "Hello, " + request.getName() + "! (sequence: " + (index + 1) + ')')
+            // You can make your Flux/Mono publish the signals in the RequestContext-aware executor.
+            .publishOn(Schedulers.fromExecutor(RequestContext.current().contextAwareExecutor()))
+            .subscribe(message -> {
+                           // Ensure this callback is executed in the RequestContext-aware executor.
+                           final ServiceRequestContext ctx = RequestContext.current();
+                           responseObserver.onNext(buildReply(message));
```

Comment:
Probably better to add `assert ctx != null;`?

Reply:
I thought it but `RequestContext.current()` returns non-null context or throws an `IllegalStateException`. So the assertion seems meaningless. 🤔

Reply:
Yeah, right. :) I just thought that the readers might be suspicious because we pulled it and did nothing. But the comment explains well, so probably not need it. 😄

Reply:
Yes, I wanted to do something with it but there was nothing to do.. 😸